src/market/directory.rs

```diff
@@ -121,6 +121,19 @@ impl DirectoryServer {
             addresses,
         })
     }
+    pub fn update_directory_config(
+        &self,
+        path: &PathBuf,
+        toml_data: String,
+    ) -> std::io::Result<()> {
+        if let Some(parent) = path.parent() {
+            fs::create_dir_all(parent)?;
+        }
+        let mut file = std::fs::File::create(path)?;
+        file.write_all(toml_data.as_bytes())?;
+        file.flush()?;
+        Ok(())
+    }
```

Comment:
Remove &self. Make this an independent util function. 
It could even be a generic util function `write_toml_to_file`. That can be useful later also in other modules.